tests/unit/sagemaker/jumpstart/test_cache.py

```diff
@@ -334,30 +334,100 @@ def test_jumpstart_cache_gets_cleared_when_params_are_set(mock_boto3_client):
+                    "version": "2.0.0",
+                    "min_version": "2.49.0",
+                    "spec_key": "community_models_specs/pytorch-ic-"
+                    "imagenet-inception-v3-classification-4/specs_v2.0.0.json",
+                }
+            ]
+        )
+
+        get_object_mocked_response = {
+            "Body": botocore.response.StreamingBody(
+                io.BytesIO(bytes(mock_manifest_json, "utf-8")),
+                content_length=len(mock_manifest_json),
+            ),
+            "ETag": "etag",
+        }
+
+        mock_datetime.now.return_value = mock_now
+
+        cache1 = JumpStartModelsCache(
+            s3_bucket_name="some_bucket", s3_cache_expiration_horizon=datetime.timedelta(hours=1)
+        )
+        stubbed_s3_client1 = Stubber(cache1._s3_client)
+
+        stubbed_s3_client1.add_response("get_object", copy.deepcopy(get_object_mocked_response))
+        stubbed_s3_client1.activate()
+        cache1.get_header(model_id="pytorch-ic-imagenet-inception-v3-classification-4")
+
+        mock_datetime.now.return_value += datetime.timedelta(weeks=1)
+
+        stubbed_s3_client1.add_client_error("head_object", http_status_code=404)
+        with pytest.raises(botocore.exceptions.ClientError):
+            cache1.get_header(model_id="pytorch-ic-imagenet-inception-v3-classification-4")
+
+        cache2 = JumpStartModelsCache(
+            s3_bucket_name="some_bucket", s3_cache_expiration_horizon=datetime.timedelta(hours=1)
+        )
+        stubbed_s3_client2 = Stubber(cache2._s3_client)
+
+        stubbed_s3_client2.add_response("get_object", copy.deepcopy(get_object_mocked_response))
+        stubbed_s3_client2.activate()
+        cache2.get_header(model_id="pytorch-ic-imagenet-inception-v3-classification-4")
+
+        mock_datetime.now.return_value += datetime.timedelta(weeks=1)
+
+        stubbed_s3_client2.add_client_error("head_object", service_error_code="AccessDenied")
+        with pytest.raises(botocore.exceptions.ClientError):
+            cache2.get_header(model_id="pytorch-ic-imagenet-inception-v3-classification-4")
+
+        cache3 = JumpStartModelsCache(
+            s3_bucket_name="some_bucket", s3_cache_expiration_horizon=datetime.timedelta(hours=1)
+        )
+        stubbed_s3_client3 = Stubber(cache3._s3_client)
+
+        stubbed_s3_client3.add_response("get_object", copy.deepcopy(get_object_mocked_response))
+        stubbed_s3_client3.activate()
+        cache3.get_header(model_id="pytorch-ic-imagenet-inception-v3-classification-4")
+
+        mock_datetime.now.return_value += datetime.timedelta(weeks=1)
+
+        stubbed_s3_client3.add_client_error(
+            "head_object", service_error_code="EndpointConnectionError"
+        )
+        with pytest.raises(botocore.exceptions.ClientError):
+            cache3.get_header(model_id="pytorch-ic-imagenet-inception-v3-classification-4")
```

Comment:
optional: we may want to create a class that extends `unittest.caseCase` and break down these tests into several class methods.